acceptance/acceptance_test.go

```diff
@@ -718,3 +727,84 @@ func filterHeaders(h http.Header, includedHeaders []string) http.Header {
 func isTruePtr(value *bool) bool {
 	return value != nil && *value
 }
+
+// runWithTail starts cmd.Run() and concurrently tails the file.
+// For each new line that appears in the file, it calls t.Log(line).
+// When cmd.Run() finishes, it uses os.Stat(filename) to get the file’s final size,
+// and stops tailing once the file offset is greater or equal to that size.
+// The function returns the error from cmd.Run().
+func runWithTail(t *testing.T, cmd *exec.Cmd, filename string) error {
+	// finalSizeCh communicates the final file size to the tail goroutine.
+	finalSizeCh := make(chan int64, 1)
+	var wg sync.WaitGroup
+	wg.Add(1)
+
+	start := time.Now()
+
+	// Tail goroutine: open the file and read new lines as they are written.
+	go func() {
+		defer wg.Done()
+
+		f, err := openWait(filename)
+		assert.NoError(t, err)
+		defer f.Close()
+
+		reader := bufio.NewReader(f)
+		finalSize := int64(-1)
+
+		for {
+			select {
+			case fs := <-finalSizeCh:
+				finalSize = fs
+			default:
+			}
+
+			if finalSize >= 0 {
+				cur, seekErr := f.Seek(0, io.SeekCurrent)
+				assert.NoError(t, seekErr)
+				if cur >= finalSize || err != nil {
+					break
+				}
+			}
+
+			line, err := reader.ReadString('\n')
+			msg := strings.TrimRight(line, "\n")
+			if len(msg) > 0 {
+				d := time.Since(start)
+				t.Logf("%2d.%03d %s", d/time.Second, (d%time.Second)/time.Millisecond, msg)
+			}
+
+			if err != nil {
+				if errors.Is(err, io.EOF) {
+					time.Sleep(50 * time.Millisecond)
+					continue
+				}
+				t.Logf("error reading file: %v", err)
+				break
+			}
+		}
+	}()
+
+	err := cmd.Run()
+
+	fi, statErr := os.Stat(filename)
+	assert.NoError(t, statErr)
+
+	finalSizeCh <- fi.Size()
+
+	wg.Wait()
+	return err
+}
+
+func openWait(path string) (*os.File, error) {
+	deadline := time.Now().Add(1 * time.Second)
```

Comment:
Why do we need to wait for 1 second?

Reply:
We try to open the file before we launch the process which writes output.txt.

Reply:
Can't we open it after we launched the command? Instead of cmd.Run we can call cmd.Start and then cmd.Wait

Reply:
Actually I see that we create the file ourselves, so we don't need to wait for Run(). removed openWait().